### ISSUE TITLE
Iron Treads of Destruction sometimes fail Armor of the Secret Temple to proc

### ISSUE THREAD

**Issue by redledcat** [ChalkyBrush/roshpit-bug-tracker#850]

**Comment by ChalkyBrush** [comment]
I don't see how this is possible, looking at the code 🤔 
might be a specific skill it's happening with?

**Comment by redledcat** [comment]
https://www.roshpit.ca/players/91933876/heroes/209355
I can try to pay more attention, but from my observation:
It usually happens when you start cast R after period of time without any casts at all, or after teleport, resurrection, etc.
It usually does not happen when you just spam it but first or second cast can fail to trigger secret temple.
For WW with her incredible 0.25 cd on R it's not a big problem, but it possibly can affect other heroes and based on @ability4 triggers.

**Comment by GoToNightmare** [comment]
Probably you didn't cast ulti at all, if you ordered command before Iron Treads triggered, it also cause not applying R effects.

**Comment by redledcat** [comment]
@GoToNightmare nope. I certainly can see if I shoop-da-whoop giant ball of light and skill goes on cooldown.
I actually expirienced these fails pretty frequent while testing Tanari Yojimbo crash.
In crater there are a lot of cutscenes, and it usually fails when I get my control over char back after another cutscene silence. I definitely cast ultimate, but it does not apply temple buff (or refresh it as it happens usually)

**Comment by redledcat** [comment]
I think it happens always after Archivist stun when he is holding you with red ray cutscene

**Comment by redledcat** [comment]
Also one really frequent fail happens when WW touches Val'Dun electric spheres and being paralyzed.
